lib/Backend/GlobOpt.cpp

```diff
@@ -15609,16 +15665,23 @@ GlobOpt::TrackArgumentsObject()
 {
     if (PHASE_OFF(Js::StackArgOptPhase, this->func))
     {
-        this->CannotAllocateArgumentsObjectOnStack();
+        this->CannotAllocateArgumentsObjectOnStack(nullptr);
         return false;
     }
 
     return func->GetHasStackArgs();
 }
 
 void
-GlobOpt::CannotAllocateArgumentsObjectOnStack()
+GlobOpt::CannotAllocateArgumentsObjectOnStack(Func * curFunc)
 {
+    if (curFunc != nullptr && curFunc->hasArgLenAndConstOpt)
+    {
+        Assert(!curFunc->GetJITOutput()->GetOutputData()->disableStackArgOpt);
+        curFunc->GetJITOutput()->GetOutputData()->disableStackArgOpt = true;
```

Comment:
>disableStackArgOpt [](start = 50, length = 18)

Why do we have to disable the stack args opt as a whole? Can't we just disable argLenAndConstOpt (and then check for it being enabled in OptArgLenAndConst) ?